src/zeroconf/_cache.py

```diff
@@ -252,4 +302,9 @@ def async_mark_unique_records_older_than_1s_to_expire(
                 created_double = record.created
                 if (now - created_double > _ONE_SECOND) and record not in answers_rrset:
                     # Expire in 1s
-                    record.set_created_ttl(now, 1)
+                    self._async_set_created_ttl(record, now, 1)
+
+    def _async_set_created_ttl(self, record: DNSRecord, now: _float, ttl: _float) -> None:
+        """Set the created time and ttl of a record."""
+        record._set_created_ttl(now, ttl)
```

Comment:
This isn't great that we mutate existing records. We really should copy them and put the new record in the cache but we don't have a copy implementation so its the best we can do right now